scipy/spatial/transform/rotation.py

```diff
@@ -281,3 +281,115 @@ def as_rotvec(self):
             return rotvec[0]
         else:
             return rotvec
+
+    @staticmethod
+    def _make_dcm_from_angle(char, angle, intrinsic=False):
+        sinx = np.sin(angle)
+        cosx = np.cos(angle)
+
+        axis = char.lower()
+        if axis == 'z':
+            dcm = np.array([
+                [cosx, -sinx, 0],
+                [sinx, cosx, 0],
+                [0, 0, 1]
+                ])
+        elif axis == 'y':
+            dcm = np.array([
+                [cosx, 0, sinx],
+                [0, 1, 0],
+                [-sinx, 0, cosx]
+                ])
+        elif axis == 'x':
+            dcm = np.array([
+                [1, 0, 0],
+                [0, cosx, -sinx],
+                [0, sinx, cosx]
+                ])
+
+        return dcm.transpose() if intrinsic else dcm
+
+    @classmethod
+    def _make_dcm_from_euler(cls, axis_str, angle):
```

Comment:
I'm not sure if I like the approach you took here. I see the two possibilities:

1. We multiply "elementary" quaternions. In this case you need to implement quaternion multiplication routine. You can keep it as a standalone function.
2. We compute dcms in terms of trigonometric functions directly, as explained in the paper I will send to you and then convert to quaternions. Also note that the paper contains an alternative approach to extract Euler angles for an arbitrary sequence. You should definitely study it as well.